extensions/churchtools/src/hooks/useCTFetch.ts

```diff
@@ -0,0 +1,32 @@
+import { useFetch } from "@raycast/utils";
+import { ArgumentTypes } from "../types/ArgumentTypes";
+import { getPreferenceValues } from "@raycast/api";
+
+export const useCTFetch = (apiPath: string, args?: ArgumentTypes<typeof useFetch>[1]) => {
+  const { loginToken, instanceUrl } = getPreferenceValues<Preferences>();
+
+  const apiUrl = `https://${instanceUrl}/api${apiPath}`;
+
+  if (args) {
+    args.headers = {
+      ...args.headers,
+      Authorization: `Login ${loginToken}`,
+    };
+  } else {
+    args = {
+      headers: {
+        Authorization: `Login ${loginToken}`,
+      },
+    };
+  }
+
+  const response = useFetch(apiUrl, args);
+
+  if (response.data && typeof response.data === "object" && "data" in response.data) {
+    return { ...response, data: response.data.data };
+  } else if (response.data === undefined) {
+    return { ...response, data: undefined };
+  } else {
+    throw new Error("Invalid response");
+  }
```

Comment:
Is this needed? I think it's handled by useFetch automatically, right?

Reply:
Nearly all ChurchTools API Endpoints return a JSON object in the following format:
```json
{
    "data": {...}
}
``
`

And since `responce.data` includes this data object, I extract it here so not to have to write `data.data` every time I want to access the actual data.